message-processor/src/main/java/com/bosch/iot/things/example/message/processor/transport/AmqpServer.java

```diff
@@ -0,0 +1,153 @@
+    private Vertx vertx;
+
+    @Value(value = "${tenant.id}")
+    private String tenantId;
+
+    @Value(value = "${local.client.port}")
+    private Integer localPort;
+
+    private ProtonServer localServer;
+
+    private ProtonConnection connection;
+
+    private Map<String, ProtonSender> senders = new HashMap<>();
+
+    @PostConstruct
+    private void start() {
+        // Create the Vert.x AMQP server instance
+        setLocalServer(ProtonServer.create(vertx));
+        listen();
+    }
+
+    private void listen(){
+        localServer.connectHandler((connection) -> {
+            setConnection(connection);
+            handleConnection(connection);
+        }).listen(localPort,(res) -> {
+            if (res.succeeded()) {
+                log.info("Listening on: " + res.result().actualPort());
+            } else {
+                log.error("Error while starting local AMQP server!");
+                res.cause().printStackTrace();
+            }
+        });
+    }
+
+    private void handleConnection(ProtonConnection connection) {
+        connectionHandler(connection);
+        connectionReceiverHandler(connection);
+        connectionSenderHandler(connection);
+    }
+
+    public void connectionHandler(ProtonConnection connection) {
+        connection.openHandler(res -> {
+            log.info("Client connected: " + connection.getRemoteContainer());
+            connection.open();
+        }).closeHandler(c -> {
+            log.info("Client closing amqp connection: " + connection.getRemoteContainer());
+            connection.close();
+            connection.disconnect();
+        }).disconnectHandler(c -> {
+            log.info("Client socket disconnected: " + connection.getRemoteContainer());
+            connection.disconnect();
+        }).sessionOpenHandler(session -> session.open());
+    }
+
+    private void connectionReceiverHandler(ProtonConnection connection) {
+        connection.receiverOpenHandler(receiver -> {
+            log.info("Receiving from: " + receiver.getRemoteTarget().getAddress());
+            receiver.setTarget(receiver.getRemoteTarget())
+                    .handler((delivery, msg) -> {
+                        String address = receiver.getRemoteTarget().getAddress();
+                        logMessageInfo(msg, address);
+                        sendMessage(msg, address);
+                    }).open();
+        });
+    }
+
+    private void logMessageInfo(Message msg, String address) {
+        log.debug("MSG.BODY_TYPE: " + msg.getBody().getType());
+        log.debug("MSG.ADDRESS: " + msg.getAddress());
+        log.debug("MSG.REPLY_TO: " + msg.getReplyTo());
+        log.debug("MSG.SUBJECT: " + msg.getSubject());
+        log.info("message to:" + address);
+        log.debug("body: " + msg.getBody().toString());
+    }
+
+    private void sendMessage(Message msg, String address) {
+        if (address.startsWith(CONTROL_ENDPOINT) && !address.endsWith(CONTROL_REPLY_ENDPOINT)) {
```

Comment:
This would not be necessary if we wire the associated receivers/senders on creation (e.g. a receiver on Things side will never receive a telemetry or event message and a receiver on hub side will never receive a command message).